### PR TITLE
Fix duplicate menu key keyboard entries

### PR DESCRIPTION
**Pull request by apartfromtime**
Fixes a bug where duplicate keyboard entries would occur in menu fields when using the keypad keys with num-lock on.
Menu fields now mirror behaviour of the console field.
Removed keypad left arrow case from console key ignore list.

### CODE REVIEW

**Comment by Yamagi** [comment]
Hi,

that is nice find :) This solution has some unwanted implications, though. First, the code distinguishes between character events and key events because in SDL 1.2 and most likely in the Vanilla Quake 2 backends for Win9x and X11 these were different kind of events. SDL2 got away with that and has only key events. In a good world YQ2 would use only key events and no character events, text input would be implemented through SDL2s input events. Input events are an abstraction over key events. In practice no one ever had the motivation to modernize the code accordingly and most likely it wouldn't be a good idea either. The chance to introduce new bugs or uncover hidden corner cases is extremely high.

Your approach alters the behavior of the numpad for the whole game. The key bindings are now dependent on the state of the num key which at least messes with binding and may break some cases like numblocks embedded into the normal part of the keyboard or some strange keyboard layout.

And the console behavior gets inconsistent. Numpad left is now allowed in the console, numpad up and down aren't. I would prefer not to mess with the console because the console code is broken, fragile and the change has potential to uncover hidden bugs or design limitations.

What about another approach: Add a function `Sys_NumpadIsOn()` or something like that to the input backend and leave it otherwise alone. Use that function to fix the menu inside the menu code.

**Comment by apartfromtime** [comment]
`Key_GetMenuKey()` introduced a subtle bug by remapping inputs. I've implemented your suggestion, this fixes the duplicate entries, the keypad directional keys work in the menu as one would expect and any key-binds, bind the keypad key not the numpad character.

File: src/client/menu/menu.c [hunk 1]
```diff
@@ -283,12 +287,18 @@ Key_GetMenuKey(int key)
 		case K_BACKSPACE:
 		case K_DEL:
 		case K_KP_DEL:
+                    if (IN_NumpadIsOn() == true) { break; }
```
Comment (Yamagi) [inline src/client/menu/menu.c:290]: Not 100% sure but it seems like the indention is wrong.

File: src/client/menu/qmenu.c [hunk 2]
```diff
@@ -223,51 +223,31 @@ extern int keydown[];
...
+        case K_KP_5:
+        case K_KP_RIGHTARROW:
+        case K_KP_END:
+        case K_KP_DOWNARROW:
+        case K_KP_PGDN:
+        case K_KP_INS:
+        case K_KP_DEL:
+        default:
+            break;
+    }
```
Comment (Yamagi) [inline src/client/menu/qmenu.c:250]: The switch-case statement is now superfluous and can be deleted.

**Review by Yamagi — CHANGES_REQUESTED** [review]
This looks now much better. Thank you for your changes. I have still two smaller points. I think that it it's ready for merging after these have been resolved.

File: src/client/menu/menu.c (hunk 1 above)
Comment (apartfromtime) [reply src/client/menu/menu.c]: Weird one, displays correctly in the editors, Notepad, Sublime Text, Visual Studio, but tabs are broken when posting to GitHub. Very annoying. Vim resolved this.

File: src/client/menu/qmenu.c (hunk 2 above)
Comment (apartfromtime) [reply src/client/menu/qmenu.c]: Deleted.

**Review by Yamagi — APPROVED** [review]

**Comment by Yamagi** [comment]
Now it perfekt. Thank you for your hard work!

**Pull request merged**